### PR TITLE
feat: replace API polling with db websocket sub

### PR DESCRIPTION
**Pull request by 0xernesto**

### CODE REVIEW

File: src/utils/supabase/client.ts [hunk 1]
```diff
@@ -3,37 +3,49 @@ import { createClient } from "@supabase/supabase-js";
...
+// 				return fetch(url, {
+// 					...options,
+// 					headers,
+// 				});
+// 			},
+// 		},
+// 	});
+// }
+
+// NOTE: The above errors out. This, along with disabling RLS in Supabase, works.
```
Comment (marvinmarnold) [inline src/utils/supabase/client.ts:46]: This code is pretty important. It makes so users can only access DB entries that belong to them. Did you run the migrations on your local supabase?

I think the way things are now, DB entries are publicly accessible. But I should be able to figure this out if you don't have time.

Comment (0xernesto) [reply src/utils/supabase/client.ts:46]: Makes sense. I'll look further into it. One thing that comes to mind is that the Clerk token expires very quickly, so it makes sense to pass a fresh token as a parameter to `getSupabase()`. The fresh token would come from `getToken()` in the component that needs the supabase client, like `HomePage`.

As long as the user remains authenticated within the app, `getToken()` will return a fresh, valid token for the current user.

File: src/app/home/page.tsx [hunk 2]
```diff
@@ -66,50 +46,142 @@ function HomePage() {
...
+			}
 		}
 	};
 
-	// Fetch lockers every 5 seconds if lockers is null
+	// Handle token transaction updates from Supabase subscription
+	// eslint-disable-next-line @typescript-eslint/no-explicit-any
+	const handleTxUpdates = async (payload: any) => {
+		console.log("Tx Change received!", payload);
+		// TODO: finish implementing this
```
Comment (marvinmarnold) [inline src/app/home/page.tsx:87]: Looks unimplemented

**Review by marvinmarnold — COMMENTED** [review]
@0xernesto I just tried to run this locally, connected to my personal Supabase, but new txs aren't being displayed via realtime. I turned off RLS. I also tried on the vercel preview link, but same problem.

Is this working in staging for you?

File: src/app/home/page.tsx (hunk 2 above)
Comment (0xernesto) [reply src/app/home/page.tsx:87]: It looks like it is an async state issue. The `lockers` state variable is still null on the first render, so the if-statement that triggers tx subscription is not executing. I'll get this fixed.

Comment (0xernesto) [reply src/app/home/page.tsx:87]: Should be fixed now

Comment (0xernesto) [reply src/app/home/page.tsx:87]: Should be working now

**Review by marvinmarnold — COMMENTED** [review]
@0xernesto I ended up doing a major rework of this. It did rely on open permissions and  the data fetching was pretty hard for me to follow. I did also encounter issues getting RLS to play nice with Clerk but this thread was very helpful: https://github.com/supabase/supabase-js/issues/553#issuecomment-1924720347

I'll do a bit more testing in staging and then will merge this soon. Thanks for all the help.

**Pull request merged**